### DISCUSSION TITLE
Available value list of Accelerator.barrier's `name` argument

### DISCUSSION THREAD

**Discussion by tarepan** [Lightning-AI/pytorch-lightning#5787] [Idea pool]
## ❓ Questions and Help
Are there the list of available `Accelerator.barrier`'s `name` argument value?  

#### What is your question?
`Accelerator.barrier` function accept `name` argument ([doc](https://pytorch-lightning.readthedocs.io/en/stable/accelerators.html#implement-a-custom-accelerator))  
Type of `name` is `Optional[str]`, so there is no information of available `str` value.  
Are there available value list?  

By the way, in `CheckpointConnector`, `name='TrainerIOMixin.restore_weights'` is used.  
There is no module named `TrainerIOMixin`, and `restore_weights` is hard coded function name (it is now too old name...).  
If the list is clarified, we can refactor this not good `name` value.  

#### Code
from doc:

```python
class MyAccelerator(Accelerator):
  ...
  def barrier(self, name: Optional[str] = None):
            # implement this accelerator's barrier
```
`name` is `Optional[str]`.  

#### What's your environment?
master f7402455218e5087e61d8255a4a87a8db58a7194

**Comment by carmocca** [comment]
The name argument is just for readability. As a comment for why the barrier is performed.

**Reply by tarepan** [reply]
Great info, thanks!
